### DISCUSSION TITLE
antdV5 Menu如何对选中的SubMenu动态添加class

### DISCUSSION THREAD

**Discussion by eveforest** [ant-design/ant-design#42108] [Q&A] [answered]
Menu 的children提示Deprecated，但使用items={menus}的方式，我没有找到有什么方法能对选中的标签添加class来自定义样式，有什么方法可以做到呢？

**Comment by BoyYangzai** [comment]
@eveforest 
![image](https://user-images.githubusercontent.com/94534613/235821202-246efa6a-8a41-49dd-aff8-15b95eb0dce5.png)
或许可以通过改变`ant-menu-item-selected`这一个className的样式来做到

**Reply by eveforest** [reply]
可是collapsed=true状态下，选中的是item，ant-menu-item-selected这个class在menuitem上，submenu上没有变化，只能想办法在事件中为submenu添加自定义class，只是我无法获取到选中item时的submenu元素

**Reply by BoyYangzai** [reply]
@eveforest 
指的是这个元素吗
![image](https://user-images.githubusercontent.com/94534613/235833569-be33fbcb-1228-4a80-8614-e28bda858dcc.png)
如果是的话 通过这两个className应该可以达到预期
![image](https://user-images.githubusercontent.com/94534613/235833545-a0c9b56f-f838-4e63-9b7e-6c8fe91395a3.png)

**Reply by eveforest** [reply]
对，但是collapsed下没有ant-menu-submenu-open

**Reply by BoyYangzai** [reply]
@eveforest 
应该是这个效果吧
![image](https://user-images.githubusercontent.com/94534613/235834456-fab3387d-9194-4bdf-a49f-c29bb77c94fb.png)
`ant-menu-submenu-selected` - `ant-menu-submenu-title` 你看看这两个符合预期吗 还是我理解有问题
![image](https://user-images.githubusercontent.com/94534613/235834497-7266394f-3d16-411c-a6ff-6442127ceca3.png)

**Reply by eveforest** [reply]
你收起来的再看看class，没有selected和open了

**Reply by BoyYangzai** [reply]
@eveforest 上面这个是收起来的状态了 你看看右边的箭头icon

**Reply by eveforest** [reply]
噢，是sider收起来，就是折叠噢，是sider收起来，就是折叠

**Reply by BoyYangzai** [reply]
> 噢，是sider收起来，就是折叠噢，是sider收起来，就是折叠

不太理解了哥们 可以的话给个sandbbox的复现demo吧

**Reply by eveforest** [reply]
![image](https://user-images.githubusercontent.com/32702981/235835721-e228859c-a243-4730-b5d0-51f3855d5eb8.png)

![image](https://user-images.githubusercontent.com/32702981/235835680-772f44b0-20ed-47c8-8178-6a454e2a8ef3.png)
折叠状态下，submenu的class在选择后就没有open等效果class，我想要在选择item后的submenu保持选中样式

**Reply by BoyYangzai** [reply]
@eveforest 你是不是在折叠之后立马取消了对submenu中的item的选中状态
下图中是有`ant-menu-submenu ant-menu-submenu-vertical ant-menu-submenu-selected`这几个class的
![image](https://user-images.githubusercontent.com/94534613/235836718-4ef2c150-129c-4af7-bcde-878f2a7a96ae.png)

**Reply by eveforest** [reply]
没有，是bug吗，我用的版本是5.3.3

**Reply by eveforest** [reply]
我是控制在layout sider控制的collapsed

**Reply by BoyYangzai** [reply]
@eveforest 啧 这么奇怪吗 你看看你改变collapsed的时候有没有取消对于submenu的item的选中状态
在这个过程中submenu的item的选中状态是不变的

**Reply by eveforest** [reply]
@BoyYangzai 我更新版本后正常了

**Reply by BoyYangzai** [reply]
> @BoyYangzai 我更新版本后正常了

辛苦了😄刚健完身回来doge

**Reply by eveforest** [reply, marked as answer]
@BoyYangzai 不好意思，不是版本的问题，是我使用了重复的key，那个class加到别的submenu去了，抱歉抱歉

**Reply by BoyYangzai** [reply]
> @BoyYangzai 不好意思，不是版本的问题，是我使用了重复的key，那个class加到别的submenu去了，抱歉抱歉

哈哈哈好的好的 下次再有别人问 我让他们注意一下这个问题😄

**Comment by eveforest** [comment]
```
        <Sider
            className={styles.siderNav}
            collapsible
            // trigger={null}
            collapsed={collapsed}
            collapsedWidth={'4em'}
        >
            <Menu
                ref={menuRef}
                className={clsNames}
                theme="dark"
                mode="inline"
                defaultSelectedKeys={[newMenuList[0]?.key as string]}
                items={newMenuList}
                onSelect={handleSelect}
            >
            </Menu>
```
```
.ant-menu {

    .ant-menu-item,
    .ant-menu-submenu-title {
        margin-inline: 0;
        margin-block: 0;
        width: 100%;
    }

    .ant-menu-submenu-arrow {
        right: 2em;
    }
}
.ant-menu-inline-collapsed {
    .ant-menu-item-selected {
        background-color: unset;
    }
    @collapsedSelectedEffect: {
        position: absolute;
        top: 0;
        left: 0;
        width: 3px;
        height: 100%;
        content: "";
        background-color: #0960bd;
    }
    .ant-menu-item-selected:before {
        @collapsedSelectedEffect();
    }
}
```
我的代码很简单，没有做什么复杂操作，甚至不知道如何通过只配置items的方式取消选中状态

**Reply by eveforest** [reply]
@BoyYangzai 你看看代码是不是有什么问题
